### PR TITLE
Testing To Do list: part 2

### PR DESCRIPTION
**Pull request by GedeonTS**
# Testing 
In this part with my partner @tugsem, we :
- Created a test file for files containing the following functions:
  -  for editing the task description.
  - for updating an item's 'completed' status
  - the "Clear all completed" function.

### CODE REVIEW

File: Test2.test.js [hunk 1]
```diff
@@ -0,0 +1,80 @@
+/**
+ * @jest-environment jsdom
```
Comment (Ismailco) [inline Test2.test.js]: - [x] The test is not working due to jsdom not being configured correctly, please fix it (You may need to install jsdom using npm)

Screenshot_20220506_135700

**Review by Ismailco — CHANGES_REQUESTED** [review]
### Hi @GedeonTS, @tugsem,

Good job so far 👍🏼 
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights

✔️ No linter errors
✔️ Tests pass
✔️ Work documented professionally

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: Test2.test.js (hunk 1 above)
Comment (tugsem) [reply Test2.test.js]: 👍

File: Test2.test.js [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+  const addAndRemove3 = new UserTask('vitron3', false, 6);
+  addAndRemove3.add();
+  const addAndRemove4 = new UserTask('vitron4', false, 7);
+  addAndRemove4.add();
+  const addAndRemove5 = new UserTask('vitron4', false, 8);
+  addAndRemove5.add();
+  const addAndRemove6 = new UserTask('vitron4', false, 9);
+  addAndRemove6.add();
+  const addAndRemove7 = new UserTask('vitron4', false, 10);
+  addAndRemove7.add();
```
Comment (gtekle) [inline Test2.test.js:34]: - [ Optional] It is good to use [setup and tear down](https://jestjs.io/docs/setup-teardown) jest methods to prepare data for all/each test case and clear after all/each test case. Please take a look the jest documentation following [this](https://jestjs.io/docs/setup-teardown) link.

**Review by gtekle — APPROVED** [review]
Hi **@TEAM**, 
# STATUS: APPROVED :clap: :clap: :clap:
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
approved-stamp-02


## Highlights
:heavy_check_mark: Professional documentation
:heavy_check_mark: Descriptive pull request summary

## Optional suggestions

- [ Optional ] Is it correct if I say the `Task2.js` file you are testing is not used for the real Todo application? I am saying this because the `Task2.js` file and the `__mock__` folder is not included in the `index.js` file. Instead you are using the files under the modules folder. Therefore, if that is the case, I strongly suggest that you write unit tests for the logic that you actually use.

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**